src/components/network-map-viewer/network/map-equipments.ts

```diff
@@ -184,28 +174,32 @@ export class MapEquipments {
         }
     }
 
-    removeBranchesOfVoltageLevel(branchesList, voltageLevelId) {
+    removeBranchesOfVoltageLevel(branchesList: Line[], voltageLevelId: string) {
         const remainingLines = branchesList.filter(
             (l) => l.voltageLevelId1 !== voltageLevelId && l.voltageLevelId2 !== voltageLevelId
         );
-        branchesList.filter((l) => !remainingLines.includes(l)).map((l) => this.linesById.delete(l.id));
+        branchesList.filter((l) => !remainingLines.includes(l)).forEach((l) => this.linesById.delete(l.id));
 
         return remainingLines;
     }
 
-    removeEquipment(equipmentType, equipmentId) {
+    removeEquipment(equipmentType: EQUIPMENT_TYPES, equipmentId: string) {
         switch (equipmentType) {
             case EQUIPMENT_TYPES.LINE: {
                 this.lines = this.lines.filter((l) => l.id !== equipmentId);
                 this.linesById.delete(equipmentId);
                 break;
             }
             case EQUIPMENT_TYPES.VOLTAGE_LEVEL: {
-                const substationId = this.voltageLevelsById.get(equipmentId).substationId;
-                let voltageLevelsOfSubstation = this.substationsById.get(substationId).voltageLevels;
-                voltageLevelsOfSubstation = voltageLevelsOfSubstation.filter((l) => l.id !== equipmentId);
-                this.substationsById.get(substationId).voltageLevels = voltageLevelsOfSubstation;
-
+                const substationId = this.voltageLevelsById.get(equipmentId)?.substationId;
+                if (substationId === undefined) {
+                    return;
+                }
+                const substation = this.substationsById.get(substationId);
+                if (substation === undefined) {
```

Comment:
Checks if substation is either null or undefined

```suggestion
                if (substation == null) {
```

Reply:
Done